test/known_problems/should_pass/nested_pattern_match.erl

```diff
@@ -0,0 +1,12 @@
+-module(nested_pattern_match).
+-compile([debug_info]).
+-export([f/1]).
+
+-record(s, {a :: c | d}).
+-record(r, {a :: #s{} | boolean()}).
+
+-spec f(#s{}) -> boolean().
+f(R) -> if
+    (R#r.a)#s.a == c -> true;
```

Comment:
I must be missing something because I don't understand why this should type check. In the spec, the argument is declared to have type `#s{}` where the field `a` has type `c | d`. But here it is used at type `#r` where the `a` field has type `#s{} | boolean`. How is that correct?

Reply:
Whoops, `#s{}` started out as `#r{}` and then I played around and committed the wrong thing.